[Paper] Add function to calculate metrics from posteriors

SUKI-O opened the pull request:
<!-- To ensure we can review your pull request promptly please complete this template entirely. -->

<!-- Please reference the issue number here. You can replace "Fixes" with "Closes" if it makes more sense. -->
Add function to calculate metrics from posteriors from saved runs in order to avoid re-running the entire code for different metrics. 

Changes proposed in this pull request:
<!-- Please list all changes/additions here. -->
-

## Before submitting

<!-- Please complete this checklist BEFORE submitting your PR to speed along the review process. -->
- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

<!-- Please complete this checklist AFTER submitting your PR to speed along the review process. -->
- [ ] All GitHub Actions jobs for my pull request have passed.

SUKI-O committed:
init

SUKI-O committed:
fix bug

adam2392 commented on sktree/tests/test_utils.py:
Here I would test with 

- `FeatureImportanceClassifier` with the HonestForest setup we have (i.e. MIGHT, or CoMIGHT)
- Something else like SVM from sklearn

Note that posteriors returned from something like sklearn has a diff shape.

adam2392 replied:
You can test against metric functions that take in "y_score":

- http://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html
- https://scikit-learn.org/stable/modules/generated/sklearn.metrics.top_k_accuracy_score.html#sklearn.metrics.top_k_accuracy_score
- our internal implementation of mutual information

A positive test is just a "smoke test" to make sure it runs and the value is "expected". Then a negative test could be where you change something that you can foresee being a common error-surface in our workflow and make sure we error-check properly.

adam2392 committed:
Merge branch 'main' into posterior_metrics

YuxinB committed:
Update utils.py

Based on scikit-learn:
The label of the positive class. When pos_label=None, if y_true is in {-1, 1} or {0, 1}, pos_label is set to 1. Otherwise an error will be raised.
Hence, add a condition to decide the value of "pos_label"
